src/main/java/com/microsoft/sqlserver/jdbc/ConfigRetryRule.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+
+
+/**
+ * The ConfigRetryRule object is what is used by the ConfigurableRetryLogic class to handle statement retries. Each
+ * ConfigRetryRule object allows for one rule.
+ *
+ */
+public class ConfigRetryRule {
+    private String retryError;
+    private String operand = "+";
+    private int initialRetryTime = 0;
+    private int retryChange = 2;
+    private int retryCount = 1;
+    private String retryQueries = "";
+    private final String PLUS_SIGN = "+";
+    private final String MULTIPLICATION_SIGN = "*";
+    private final String COMMA = ",";
+    private final String ZERO = "0";
+
+    private ArrayList<Integer> waitTimes = new ArrayList<>();
+
+    /**
+     * Default constructor
+     *
+     * @param rule
+     *        The rule used to construct the ConfigRetryRule object
+     * @throws SQLServerException
+     *         If there is a problem parsing the rule
+     */
+    public ConfigRetryRule(String rule) throws SQLServerException {
+        addElements(parse(rule));
+        calculateWaitTimes();
+    }
+
+    /**
+     * Allows constructing a ConfigRetryRule object from another ConfigRetryRule object. Used when the first object has
+     * multiple errors provided. We pass in the multi-error object and create 1 new object for each error in the initial
+     * object.
+     *
+     * @param rule
+     *        The rule used to construct the ConfigRetryRule object
+     * @param base
+     *        The ConfigRetryRule object to base the new objects off of
+     */
+    public ConfigRetryRule(String rule, ConfigRetryRule base) {
+        copyFromExisting(base);
+        this.retryError = rule;
+    }
+
+    private void copyFromExisting(ConfigRetryRule base) {

```

Comment:
this is just a copy, specifically a shallow copy. How caller uses this shouldn't be part of the name

Reply:
the variable name base is also irrelevant, it's just a rule